File: src/go/repositories/usersRepository.go
package repositories

import (
	"app/models"
	"context"
	"errors"
	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
	"log"
	"os"
)

func GetUser(id int) (*models.User, error) {

	var client = CreateSession()
	coll := client.Database("mongo").Collection("users")
	var user models.User
	err := coll.FindOne(context.TODO(), bson.D{{"_id", id}}).Decode(&user)
	if errors.Is(err, mongo.ErrNoDocuments) {
		return nil, err
	}
	if err != nil {
		panic(err)
	}

	if err := client.Disconnect(context.TODO()); err != nil {
		panic(err)
	}

	user.Id = id
	return &user, nil
}

func CreateSession() *mongo.Client {
	uri := os.Getenv("CONNECTION_STRING")
	if uri == "" {
		log.Fatal("CONNECTION_STRING not set")
	}
	client, err := mongo.Connect(context.TODO(), options.Client().ApplyURI(uri))
	if err != nil {
		panic(err)
	}

	return client
}

File: src/rust/src/repositories/users_repository.rs
use mongodb::{
    bson::doc,
    error::Error,
    Client, 
    options::ClientOptions
};
use crate::models::user::User;

pub async fn get_user(id: u32) -> Result<User, Error> {
    let mut client_options = ClientOptions::parse("mongodb://admin:admin@host.docker.internal:27017/?directConnection=true").await?;
    let client = Client::with_options(client_options)?;
    let usersCollection = client.database("mongo").collection("users");
    let user: User = usersCollection.find_one(doc! {"_id": id}, None)
                                    .await
                                    .expect("Couldn't find user")
                                    .expect("Couldn't deserialize user");
                                           
    Ok(user)
}
